services/app-api/src/handlers/audit_s3.ts

```diff
@@ -0,0 +1,168 @@
+            'Init Error: SECRETS_MANAGER_SECRET is required to run postgres migrate'
+        recordException(errMsg, serviceName, 'secretsManagerSecret')
+        return fmtAuditError(errMsg)
+    }
+
+    if (!stage) {
+        const errMsg = 'Init Error: STAGE not set in environment'
+        recordException(errMsg, serviceName, 'stage')
+        return fmtAuditError(errMsg)
+    }
+
+    const s3Client = new S3Client({ region: 'us-east-1' })
+
+    const pgResult = await configurePostgres(dbURL, secretsManagerSecret)
+    if (pgResult instanceof Error) {
+        console.error("Init Error: Postgres couldn't be configured")
+        throw pgResult
+    }
+    const store = NewPostgresStore(pgResult)
+
+    // fetch all the documents in the db
+    const docResult = await store.findAllDocuments()
+    if (docResult instanceof Error) {
+        console.error('Audit failed:', docResult.message)
+        return {
+            statusCode: 500,
+            body: JSON.stringify(docResult),
+        }
+    }
+
+    // any doc we can't get on s3 or causes an error we care about
+    const missingOrErrorDocuments: AuditDocument[] = []
+    for (const doc of docResult) {
+        const processResult = await processDocument(s3Client, doc)
+        if (processResult) {
+            missingOrErrorDocuments.push(processResult)
+        }
+    }
+    console.info(
+        `Missing ${missingOrErrorDocuments.length} of ${docResult.length} documents`
+    )
+    console.info(
+        `These documents could not be retreived from s3: ${JSON.stringify(missingOrErrorDocuments)}`
+    )
+
+    const success: APIGatewayProxyResultV2 = {
+        statusCode: 200,
+        body: JSON.stringify('testing success'),
+        headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Credentials': true,
+        },
+    }
+    return success
+}
+
+function fmtAuditError(error: string): APIGatewayProxyResultV2 {
+    return {
+        statusCode: 500,
+        body: JSON.stringify(error),
+        headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Credentials': true,
+        },
+    }
+}
+
+async function isDocumentInS3(
+    client: S3Client,
+    s3URL: string
+): Promise<boolean> {
+    try {
+        const { bucket, key } = parseS3URL(s3URL)
+        const command = new HeadObjectCommand({
+            Bucket: bucket,
+            Key: key,
+        })
+        await client.send(command)
+        return true
+    } catch (error) {
+        if (isS3ServiceException(error) && error.name === 'NotFound') {
+            return false
+        }
+        throw error
+    }
+}
+
+function parseS3URL(s3URL: string): { bucket: string; key: string } {
+    const url = new URL(s3URL)
+    const fullPath = url.pathname.slice(1) // Remove leading '/'
+    const uuidPart = fullPath.split('/')[0] // Extract the UUID part
+    const key = `allusers/${uuidPart}` // Construct the key
+
```

Comment:
Question for review: when I get back the document entries from the DB in my review branch, I'm seeing the `s3URL` returned that has a format like `s3://bucket-name/53003860-9999-7777-8888-bc113cc0b59b.pdf/response_to_questions_for_submission.pdf`, for example. Notice that the regular key UUID is there, but then also the original filename after it. Is this really how we're storing s3 URLs in the DB? I sort of figured we'd be storing a valid s3URL here.

Reply:
see s3URLEncoding.ts in common-code for tools for this. We store the s3 url as well as the filename together.